Load Store Instructions Ported to RV32 in src/builtins and macro-assembler

usmnzain opened the pull request:
Changed the Ld and Sd instructions in builtins-riscv32 and macro-assembler-riscv32 to Lw and Sw

luyahan commented:
Hi usmanzain
There is a conflict due to upstream refactor reg.
The upstream refactor reg, so `saved_regs` need bits func to get bit. 
Please resolved this conflicts.

luyahan commented:
There is a format issue:
```
Run PATH=$PATH:$GITHUB_WORKSPACE/depot_tools git cl for --diff --no-python
--- src/builtins/riscv32/builtins-riscv32.cc	(before formatting)
+++ src/builtins/riscv32/builtins-riscv32.cc	(after formatting)
@@ -18[5](https://github.com/riscv-collab/v8/runs/5965580930?check_suite_focus=true#step:12:5),[7](https://github.com/riscv-collab/v8/runs/5965580930?check_suite_focus=true#step:12:7) +1[8](https://github.com/riscv-collab/v8/runs/5965580930?check_suite_focus=true#step:12:8)5,7 @@
     __ LoadTaggedPointerField(
         func_info, FieldMemOperand(a1, JSFunction::kSharedFunctionInfoOffset));
     __ Lw(func_info,
-           FieldMemOperand(func_info, SharedFunctionInfo::kFlagsOffset));
+          FieldMemOperand(func_info, SharedFunctionInfo::kFlagsOffset));
     __ DecodeField<SharedFunctionInfo::FunctionKindBits>(func_info);
     __ JumpIfIsInRange(
         func_info,
```
please run `git cl format --no-python` and commit it

https://github.com/riscv-collab/v8/runs/5965580930?check_suite_focus=true

qjivy commented:
Hi @usmnzain , before make pull request, you can firstly do local build using ./tools/dev/gm.py riscv32.debug to get all the to be modified lines.

```
In file included from ../../src/baseline/baseline-assembler-inl.h:38:
../../src/baseline/riscv32/baseline-assembler-riscv32-inl.h:163:6: error: no member named 'Ld' in 'v8::internal::MacroAssembler'
  __ Ld(type, FieldMemOperand(map, Map::kInstanceTypeOffset));
  ~~ ^
../../src/baseline/riscv32/baseline-assembler-riscv32-inl.h:171:6: error: no member named 'Ld' in 'v8::internal::MacroAssembler'
  __ Ld(temp, operand);
  ~~ ^
../../src/baseline/riscv32/baseline-assembler-riscv32-inl.h:195:6: error: no member named 'Ld' in 'v8::internal::MacroAssembler'
  __ Ld(scratch, operand);
  ~~ ^
../../src/baseline/riscv32/baseline-assembler-riscv32-inl.h:204:6: error: no member named 'Ld' in 'v8::internal::MacroAssembler'
  __ Ld(scratch, operand);
  ~~ ^
../../src/baseline/riscv32/baseline-assembler-riscv32-inl.h:219:6: error: no member named 'Sd' in 'v8::internal::MacroAssembler'
  __ Sd(source, output);
  ~~ ^
In file included from ../../src/baseline/baseline-compiler.cc:53:
../../src/baseline/riscv32/baseline-compiler-riscv32-inl.h:44:18: error: no member named 'Sd' in 'v8::internal::MacroAssembler'
      __ masm()->Sd(kInterpreterAccumulatorRegister, MemOperand(sp, i * 8));
      ~~~~~~~~~  ^
../../src/baseline/riscv32/baseline-compiler-riscv32-inl.h:54:18: error: no member named 'Sd' in 'v8::internal::MacroAssembler'
      __ masm()->Sd(kInterpreterAccumulatorRegister, MemOperand(sp, i * 8));
      ~~~~~~~~~  ^
../../src/baseline/riscv32/baseline-compiler-riscv32-inl.h:59:18: error: no member named 'Sd' in 'v8::internal::MacroAssembler'
      __ masm()->Sd(kInterpreterAccumulatorRegister, MemOperand(sp, i * 8));
      ~~~~~~~~~  ^
8 errors generated.
```

usmnzain committed:
changed load doubleword to load word

usmnzain commented:
@qjivy I have updated the PR as per your instructions in slack. I have pulled and rebased from the RV32G branch. I will fix the build issues in upcomming commits

usmnzain committed:
ld and sd changed to lw and sw

qjivy commented on src/compiler/backend/riscv32/code-generator-riscv32.cc:
```diff
@@ -1756,7 +1756,7 @@ CodeGenerator::CodeGenResult CodeGenerator::AssembleArchInstruction(
       ASSEMBLE_ATOMIC_LOAD_INTEGER(Lw);
       break;
     case kRiscvWord64AtomicLoadUint64:
-      ASSEMBLE_ATOMIC_LOAD_INTEGER(Ld);
```
Here should not simple change ld into lw. because it's a word64 operation.

qjivy replied:
Merge it as it is then we will fix it.

qjivy commented on src/compiler/backend/riscv32/code-generator-riscv32.cc:
```diff
@@ -1768,7 +1768,7 @@ CodeGenerator::CodeGenResult CodeGenerator::AssembleArchInstruction(
       ASSEMBLE_ATOMIC_STORE_INTEGER(Sw);
       break;
     case kRiscvWord64AtomicStoreWord64:
-      ASSEMBLE_ATOMIC_STORE_INTEGER(Sd);
```
Same as the above comment

qjivy replied:
Merge it as it is then we will fix it.
